src/main/java/de/tum/in/www1/artemis/service/connectors/gitlabci/GitLabCITriggerService.java

```diff
@@ -21,14 +24,21 @@ public class GitLabCITriggerService implements ContinuousIntegrationTriggerServi
 
     private final UriService uriService;
 
-    public GitLabCITriggerService(GitLabApi gitlab, UriService uriService) {
+    private final ProgrammingExerciseBuildConfigRepository programmingExerciseBuildConfigRepository;
+
+    public GitLabCITriggerService(GitLabApi gitlab, UriService uriService, ProgrammingExerciseBuildConfigRepository programmingExerciseBuildConfigRepository) {
         this.gitlab = gitlab;
         this.uriService = uriService;
+        this.programmingExerciseBuildConfigRepository = programmingExerciseBuildConfigRepository;
     }
 
     @Override
     public void triggerBuild(ProgrammingExerciseParticipation participation, boolean triggerAll) throws ContinuousIntegrationException {
-        triggerBuild(participation.getVcsRepositoryUri(), participation.getProgrammingExercise().getBranch());
+        ProgrammingExercise programmingExercise = participation.getProgrammingExercise();
```

Comment:
Can you check where the exercise/participation is retrieved and if we can also directly load the configuration there?

Reply:
Same here, the method is used in many places, where the buildconfig isn't needed throughout the whole process. And we only fetch it if we didn't previously